include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,260 @@
+  const grpc::string pem_root_certs() const { return pem_root_certs_; }
+  const std::vector<PemKeyCertPair>& pem_key_cert_pair_list() const {
+    return pem_key_cert_pair_list_;
+  }
+  int version() const { return version_; }
+
+  /** Setter for key materials that will be called by the user. The setter
+   * transfers ownership of the arguments to the config. **/
+  void set_key_materials(grpc::string pem_root_certs,
+                         std::vector<PemKeyCertPair> pem_key_cert_pair_list);
+  void set_version(int version) { version_ = version; };
+
+ private:
+  int version_;
+  std::vector<PemKeyCertPair> pem_key_cert_pair_list_;
+  grpc::string pem_root_certs_;
+};
+
+/** The following 2 functions are exposed for testing purposes. **/
+grpc_tls_key_materials_config* c_key_materials(
+    const std::shared_ptr<TlsKeyMaterialsConfig>& config);
+
+std::shared_ptr<TlsKeyMaterialsConfig> tls_key_materials_c_to_cpp(
+    const grpc_tls_key_materials_config* config);
+
+/** TLS credential reload arguments, wraps grpc_tls_credential_reload_arg. **/
+class TlsCredentialReloadArg {
+ public:
+  TlsCredentialReloadArg();
+  TlsCredentialReloadArg(grpc_tls_credential_reload_arg arg);
+  ~TlsCredentialReloadArg();
+
+  /** Getters for member fields. The callback function is not exposed. **/
+  void* cb_user_data() const;
+  std::shared_ptr<TlsKeyMaterialsConfig> key_materials_config() const;
+  grpc_ssl_certificate_config_reload_status status() const;
+  std::shared_ptr<grpc::string> error_details() const;
+
+  /** Setters for member fields. **/
+  void set_cb_user_data(void* cb_user_data);
+  void set_key_materials_config(
+      const std::shared_ptr<TlsKeyMaterialsConfig>& key_materials_config);
+  void set_status(grpc_ssl_certificate_config_reload_status status);
+  void set_error_details(const grpc::string& error_details);
+
+  /** Calls the C arg's callback function. **/
+  void callback();
+
+ private:
+  grpc_tls_credential_reload_arg c_arg_;
+};
+
+// Exposed for testing purposes.
+int tls_credential_reload_config_c_schedule(
+    void* config_user_data, grpc_tls_credential_reload_arg* arg);
+void tls_credential_reload_config_c_cancel(void* config_user_data,
+                                           grpc_tls_credential_reload_arg* arg);
+
+/** TLS credential reloag config, wraps grpc_tls_credential_reload_config. **/
+class TlsCredentialReloadConfig {
+ public:
+  TlsCredentialReloadConfig(const void* config_user_data,
+                            int (*schedule)(void* config_user_data,
+                                            TlsCredentialReloadArg* arg),
+                            void (*cancel)(void* config_user_data,
+                                           TlsCredentialReloadArg* arg),
+                            void (*destruct)(void* config_user_data));
+  ~TlsCredentialReloadConfig();
+
+  int Schedule(TlsCredentialReloadArg* arg) const {
+    return schedule_(config_user_data_, arg);
```

Comment:
Shouldn't we also check if `schedule_` is `nullptr` or not?

Reply:
I added such a check. It's worth pointing out that the same check is not done in the C-core version, so it may be worth adding that check there as well.